File: 921-minimum-add-to-make-parentheses-valid/921-minimum-add-to-make-parentheses-valid.java
class Solution {
    public int minAddToMakeValid(String s) {
        Stack<Character> st=new Stack<Character>();
		int cnt=0;
		for (int i = 0; i <s.length(); i++) {
			if(s.charAt(i)=='(') {
				st.push(s.charAt(i)); 
			}else {
				if(st.isEmpty()) {
					cnt++;
				}
				else {
					st.pop();
				}
			}
		}
		cnt+=st.size();
		return cnt;
    }
}

File: 921-minimum-add-to-make-parentheses-valid/921-minimum-add-to-make-parentheses-valid.cpp
class Solution {
public:
    int minAddToMakeValid(string s) {
        stack<char> st;
        int counter = 0;
        for(char ch : s){
            if(ch == ')'){
                if(st.empty())  counter++;
                else    st.pop();
            }
            else    st.push(ch);
        }
        counter += st.size();
        return counter;
    }
};